### PR TITLE
Fixes and improvements 2

### PR DESCRIPTION
**Pull request by baranyaib90**
Hi,
I would like to share again my changes with everyone.
Please check them and consider to pick some of those.
My goals were:

1. Have coredumps of crashes, when using Debug build
2. Spare memory in embedded system, like OpenWRT
3. Error handling of libcurl
4. Some housekeeping in http client

That's all, I hope, you will like it.
Sincerely,
Balázs

### CODE REVIEW

**Comment by baranyaib90** [comment]
Hi @aarond10,
could you please check my pull request if you have time?
Thanks,
Balázs

**Review by aarond10 — APPROVED** [review]

**Review by aarond10 — APPROVED** [review]

**Review by aarond10 — APPROVED** [review]

File: src/https_client.c [hunk 1]
```diff
@@ -63,63 +79,59 @@ static void https_fetch_ctx_init(https_client_t *client,
                                  const char* data, size_t datalen,
                                  struct curl_slist *resolv,
                                  https_response_cb cb, void *cb_data) {
-  ctx->curl = curl_easy_init();
+  ctx->curl = curl_easy_init(); // if failes, first setopt will fail
```
Comment (aarond10) [inline src/https_client.c:81]: Typo. 
s/failes/fails/

**Review by aarond10 — APPROVED** [review]

**Review by aarond10 — APPROVED** [review]

**Review by aarond10 — APPROVED** [review]

**Review by aarond10 — APPROVED** [review]

**Review by aarond10 — APPROVED** [review]

**Comment by aarond10** [comment]
Thanks for all these clean ups. They're great!

File: src/https_client.c (hunk 1 above)
Comment (baranyaib90) [reply src/https_client.c:81]: Sorry.
I will fix it in next - maybe latest - round :)

**Pull request merged**